subprojects/docs/src/samples/userguide/antMigration/fileDeps/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,39 @@
+plugins {
+    java
+}
+
+// tag::file-deps[]
+repositories {
+    flatDir {
+        name = "libs dir"
+        dir(file("libs"))  // <1>
+    }
+}
+
+dependencies {
+    implementation(files("libs/our-custom.jar"))  // <2>
+    implementation(":log4j:1.2.8")     // <3>
+    implementation(":commons-io:2.1")  // <3>
+}
+// end::file-deps[]
+
+// tag::retrieve-deps[]
+tasks.create<Copy>("retrieveRuntimeDependencies") {
+    into("$buildDir/libs")
+    from(configurations.runtimeClasspath)
+}
+// end::retrieve-deps[]
+
+// tag::properties[]
+val tmpDistDir by extra { file("$buildDir/dist") }
+
+tasks.create<Jar>("javadocJar") {
+    from(tasks.getByName("javadoc"))  // <1>
```

Comment:
:x: can be simplified to `from(tasks.javadoc)`

Reply:
Doesn't compile I'm afraid:

> Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
                                  public val TaskContainer.javadoc: TaskProvider<Javadoc> defined in org.gradle.kotlin.dsl

From https://builds.gradle.org/viewLog.html?buildId=17546352&buildTypeId=Gradle_Check_Quick_Java8_Oracle_Windows_integTest

Reply:
Okay, looks like `tasks.` prefix isn't needed (breaks) within a `tasks {}` block. All fixed.

Reply:
This is an area where I get confused, i.e. when is it OK to use the Provider API (and associated APIs) and when do we need to use the eager ones. I'm glad that `from()` works with the configuration avoidance API, but how are users supposed to know that?

For that matter, it's not clear why the context allows for just `javadoc` rather than `tasks.javadoc`.

Reply:
Ah, on the second point I realise that Eric's changes introduced the `tasks {}` block, in which case it's clearer why you don't need `tasks.` (but not why using it breaks the build).

Reply:
Good point. I hope other parts of the API docs are as clear!